src/library/Uno.Material/Controls/Ripple.cs

```diff
@@ -0,0 +1,129 @@
+﻿/*
+ * Starting implementation acknowledgement from project https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/2740f14a814896d42032ae0013b765a8a0ec04c3/MaterialDesignThemes.Uwp/Ripple.cs
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Windows.Devices.Input;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace Uno.Material.Controls
+{
+	public partial class Ripple : ContentControl, INotifyPropertyChanged
+	{
+		private double _rippleSize;
+		private double _rippleX;
+		private double _rippleY;
+
+		public Ripple()
+		{
+			DefaultStyleKey = typeof(Ripple);
+			SizeChanged += OnSizeChanged;
+		}
+
+		private void OnSizeChanged(object sender, SizeChangedEventArgs sizeChangedEventArgs)
+		{
+			RippleSize = Math.Max(sizeChangedEventArgs.NewSize.Width, sizeChangedEventArgs.NewSize.Height) * RippleSizeMultiplier;
+			Clip = new RectangleGeometry { Rect = new Windows.Foundation.Rect(0, 0, sizeChangedEventArgs.NewSize.Width, sizeChangedEventArgs.NewSize.Height) };
+		}
+
+		public static readonly DependencyProperty FeedbackProperty = DependencyProperty.Register(
+			"Feedback", typeof(Brush), typeof(Ripple), new PropertyMetadata(default(Brush)));
+
+		public Brush Feedback
+		{
+			get { return (Brush)GetValue(FeedbackProperty); }
+			set { SetValue(FeedbackProperty, value); }
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			VisualStateManager.GoToState(this, "Normal", false);
+
+			base.OnApplyTemplate();
+		}
+
+		protected override void OnPointerPressed(PointerRoutedEventArgs e)
+		{
+			var point = e.GetCurrentPoint(this);
+
+			RippleX = point.Position.X - RippleSize / 2;
+			RippleY = point.Position.Y - RippleSize / 2;
+
+			VisualStateManager.GoToState(this, "Normal", true);
+			VisualStateManager.GoToState(this, "Pressed", true);
+
+			base.OnPointerPressed(e);
+		}
+
+		// Uncomment for hover effect
+		//protected override void OnPointerMoved(PointerRoutedEventArgs e)
```

Comment:
any reason why we disabled this? this could be added to the comment

Reply:
It's nice to have feature but not explicitly needed according the zeplin design

Reply:
If it works we could use it :)

Reply:
Oh well after looking back on zeplin. Probably best to hold off I suppose. You were right